app/models/exchange_rate_currency_rate.rb

```diff
@@ -44,6 +48,16 @@ def by_year_and_month(month, year = Time.zone.today.year)
       where { (validity_start_date >= start_of_month) & (validity_start_date <= end_of_month) }
         .order(Sequel.desc(:validity_start_date))
     end
+
+    def files_for_year_and_month(month, year = Time.zone.today.year)
+      return if month.blank? || year.blank?
+
+      start_of_month = Time.zone.parse("#{year}-#{month}-01").beginning_of_month
+      end_of_month = start_of_month.end_of_month
+
+      where { (validity_start_date >= start_of_month) & (validity_start_date <= end_of_month) }
```

Comment:
not sure here,
could we use here the "TimeMachine"?

Reply:
hmm I dont think so, i think this would be slightly different but good to raise!